### PR TITLE
Support swift v3 authentication

### PR DESCRIPTION
**Pull request by icewind1991**
- [x] requires https://github.com/nextcloud/server/pull/8259
- [x] requires https://github.com/nextcloud/server/pull/8314

Add support for using swift v3 authentication for external and primary storage

cc @oparoz

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Conflicts :/

**Comment by rullzer** [comment]
Rebased on master

**Comment by rullzer** [comment]
@icewind1991 can we please have two CI steps one that tests v2 and one that tests v3.

Also we can probably log a warning if users use V2 right since that has been deprecated since ages?

**Comment by rullzer** [comment]
A lot of my changes can be removed or squased later. It looks like we need a bit smarter way to detect how to find out v2 or v3 is used

**Comment by rullzer** [comment]
I *think* (tm) this works now. I'm not sure if we maybe should make it explicit which version is used in the config?

I also want CI to pass on this (but the one node running seems to not like swift). But I really don't want to break stuff here (we did that before with object storages so please not again).

**Comment by rullzer** [comment]
So it seems for some dark reason on some drone hosts this does :boom: on others it works as expected.

**Comment by rullzer** [comment]
One succesfull v3 run! https://drone.nextcloud.com/nextcloud/server/5122/248

**Review by rullzer — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]

File: config/config.sample.php [hunk 1]
```diff
@@ -1214,6 +1214,28 @@
...
+'objectstore' => [
+	'class' => 'OC\\Files\\ObjectStore\\Swift',
+	'arguments' => [
+		'autocreate' => true,
+		'user' => [
+			'name' => 'swift',
+			'password' => 'swift',
+			'domain' => [
+				'name' => 'default',
+			]
```
Comment (oparoz) [inline config/config.sample.php]: @rullzer missing comma?

Comment (MorrisJobke) [reply config/config.sample.php]: This is optional for the last element of an array (we usually add it to make the possible diff better readable.

**Comment by scysys** [comment]
Merged this in my installation and this solved v3 issues on my side too!

**Pull request merged**